### ISSUE TITLE
Layout frozen with Optional Header

### ISSUE THREAD

**Issue by yyjdelete** [icsharpcode/ILSpy#2421]
I'm not sure if it's an bug of layout of ILSpy UI or common issue of wpf

### Steps to reproduce
1. Open Search and Analyze tab, to make sure the size is prefect to reproduce the issue;
2. Click `Metadata/Optional Header` with any assembly;
3. Drop the scrollbar to the bottom;
>  Now you are unable to see the children table;
![image](https://user-images.githubusercontent.com/1819074/119809504-c5e1e180-bf17-11eb-9fb2-929f36ad9d63.png)
4. Click the line, press cursor key `UP`;

### Error message shown
UI of ILSpy frozen for more than 1 minute with single core CPU 100%.

### Details
* Product in use: ILSpy 
* Version in use: 7.0.0

**Comment by siegfriedpammer** [comment]
Sorry, I cannot reproduce this.

**Comment by yyjdelete** [comment]
I don't why is not reproducable, may be an special size is required.

1. The right scrollbar need to be drag to end, and ensure `Loader Flags` or `Number of Data Directories` is the first line; If it's not the first, try to open Search and Analyze  tab or resize the window(It happen for me with 1920*1080, max window, open Search and Analyze tab)
2. choose the first line(`Loader Flags`), and press cursor `UP` key;(the pre one is the children table)
3. It will start eat CPU and memory for more than 1 minute;

And the memory will keep grow when CPU is single core 100%
```
System.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.

   at System.Windows.Controls.Decorator.get_LogicalChildren()

   at System.Windows.DescendentsWalker`1.WalkFrameworkElementLogicalThenVisualChildren(FrameworkElement feParent, Boolean hasLogicalChildren)

   at System.Windows.DescendentsWalker`1.IterateChildren(DependencyObject d)

   at System.Windows.DescendentsWalker`1._VisitNode(DependencyObject d, Boolean visitedViaVisualTree)

   at System.Windows.DescendentsWalker`1.WalkFrameworkElementLogicalThenVisualChildren(FrameworkElement feParent, Boolean hasLogicalChildren)

   at System.Windows.DescendentsWalker`1.IterateChildren(DependencyObject d)

   at System.Windows.DescendentsWalker`1.StartWalk(DependencyObject startNode, Boolean skipStartNode)

   at System.Windows.FrameworkElement.OnPropertyChanged(DependencyPropertyChangedEventArgs e)

   at System.Windows.DependencyObject.NotifyPropertyChange(DependencyPropertyChangedEventArgs args)

   at System.Windows.DependencyObject.UpdateEffectiveValue(EntryIndex entryIndex, DependencyProperty dp, PropertyMetadata metadata, EffectiveValueEntry oldEntry, EffectiveValueEntry& newEntry, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType)

   at System.Windows.DependencyObject.SetValueCommon(DependencyProperty dp, Object value, PropertyMetadata metadata, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType, Boolean isInternal)

   at System.Windows.Controls.ItemContainerGenerator.LinkContainerToItem(DependencyObject container, Object item)

   at System.Windows.Controls.ItemContainerGenerator.OnItemReplaced(Object oldItem, Object newItem, Int32 index)

   at System.Windows.Controls.ItemContainerGenerator.OnCollectionChanged(Object sender, NotifyCollectionChangedEventArgs args)

   at System.Windows.WeakEventManager.ListenerList`1.DeliverEvent(Object sender, EventArgs e, Type managerType)

   at System.Windows.WeakEventManager.DeliverEvent(Object sender, EventArgs args)

   at System.Collections.Specialized.NotifyCollectionChangedEventHandler.Invoke(Object sender, NotifyCollectionChangedEventArgs e)

   at System.Windows.Data.CollectionView.OnCollectionChanged(NotifyCollectionChangedEventArgs args)

   at System.Windows.WeakEventManager.ListenerList`1.DeliverEvent(Object sender, EventArgs e, Type managerType)

   at System.Windows.WeakEventManager.DeliverEvent(Object sender, EventArgs args)

   at System.Windows.Data.CollectionView.OnCollectionChanged(NotifyCollectionChangedEventArgs args)

   at System.Windows.Data.ListCollectionView.ProcessCollectionChangedWithAdjustedIndex(NotifyCollectionChangedEventArgs args, Int32 adjustedOldIndex, Int32 adjustedNewIndex)

   at System.Windows.Controls.MultipleCopiesCollection.OnCollectionChanged(NotifyCollectionChangedEventArgs e)

   at System.Windows.Controls.MultipleCopiesCollection.set_CopiedItem(Object value)

   at System.Windows.Controls.Primitives.DataGridCellsPresenter.OnItemChanged(Object oldItem, Object newItem)

   at System.Windows.Controls.DataGridRow.OnItemChanged(Object oldItem, Object newItem)

   at System.Windows.Controls.DataGridRow.NotifyPropertyChanged(DependencyObject d, String propertyName, DependencyPropertyChangedEventArgs e, DataGridNotificationTarget target)

   at System.Windows.Controls.DataGridRow.OnNotifyRowPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)

   at System.Windows.DependencyObject.OnPropertyChanged(DependencyPropertyChangedEventArgs e)

   at System.Windows.FrameworkElement.OnPropertyChanged(DependencyPropertyChangedEventArgs e)

   at System.Windows.Controls.DataGridRow.OnPropertyChanged(DependencyPropertyChangedEventArgs e)

   at System.Windows.DependencyObject.NotifyPropertyChange(DependencyPropertyChangedEventArgs args)

   at System.Windows.DependencyObject.UpdateEffectiveValue(EntryIndex entryIndex, DependencyProperty dp, PropertyMetadata metadata, EffectiveValueEntry oldEntry, EffectiveValueEntry& newEntry, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType)

   at System.Windows.DependencyObject.SetValueCommon(DependencyProperty dp, Object value, PropertyMetadata metadata, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType, Boolean isInternal)

   at System.Windows.DependencyObject.SetValue(DependencyProperty dp, Object value)

   at System.Windows.Controls.DataGridRow.ClearRow(DataGrid owningDataGrid)

   at System.Windows.Controls.ItemsControl.MS.Internal.Controls.IGeneratorHost.ClearContainerForItem(DependencyObject container, Object item)

   at System.Windows.Controls.ItemContainerGenerator.UnlinkContainerFromItem(DependencyObject container, Object item, IGeneratorHost host)

   at System.Windows.Controls.ItemContainerGenerator.Remove(GeneratorPosition position, Int32 count, Boolean isRecycling)

   at System.Windows.Controls.VirtualizingStackPanel.CleanupRange(IList children, IItemContainerGenerator generator, Int32 startIndex, Int32 count)

   at System.Windows.Controls.VirtualizingStackPanel.CleanupContainers(Int32 firstItemInExtendedViewportIndex, Int32 itemsInExtendedViewportCount, ItemsControl itemsControl, Boolean timeBound, Int32 startTickCount)

   at System.Windows.Controls.VirtualizingStackPanel.MeasureOverrideImpl(Size constraint, Nullable`1& lastPageSafeOffset, List`1& previouslyMeasuredOffsets, Nullable`1& lastPagePixelSize, Boolean remeasure)

   at System.Windows.Controls.VirtualizingStackPanel.MeasureOverride(Size constraint)

   at System.Windows.Controls.Primitives.DataGridRowsPresenter.MeasureOverride(Size constraint)

   at System.Windows.FrameworkElement.MeasureCore(Size availableSize)

   at System.Windows.UIElement.Measure(Size availableSize)

   at System.Windows.ContextLayoutManager.UpdateLayout()

   at System.Windows.Interop.HwndSource.Process_WM_SIZE(UIElement rootUIElement, IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam)

   at System.Windows.Interop.HwndSource.LayoutFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)

   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)

   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)

   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)

   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

**Comment by siegfriedpammer** [comment]
I was able to reproduce this. I tried fixing this by refactoring our code, so it does no longer allocate as much objects as it did before. However, this looks like an issue in WPF. It only happens if you resize the data grid so that it needs a scrollbar and then keep the down arrow pressed for an extended period of time. Even if you do that it stops allocating objects at some point. I don't know what is causing this and it is very strange.
